src/createAction.js

```diff
@@ -10,35 +9,30 @@ export default function createAction(type, payloadCreator = identity, metaCreato
     'Expected payloadCreator to be a function, undefined or null'
   );
 
-  const finalPayloadCreator = isNull(payloadCreator)
+  const finalPayloadCreator = payloadCreator === null || payloadCreator === identity
     ? identity
-    : payloadCreator;
+    : (head, ...args) => head instanceof Error ?
+        head : payloadCreator(head, ...args);
+  
+  const hasMeta = isFunction(metaCreator);
 
   const actionCreator = (...args) => {
-    const hasError = args[0] instanceof Error;
-
-    const action = {
-      type
-    };
-
-    const payload = hasError ? args[0] : finalPayloadCreator(...args);
-    if (!isUndefined(payload)) {
+    const payload = finalPayloadCreator(...args);
+    const action = { type, error: payload instanceof Error };
+    
+    if (payload !== undefined) {
```

Comment:
It's not a readability thing; just avoiding the function call overhead. I figured that the purpose of `isUndefined` is to be used as a predicate, like in `filter`, but there's no reason to use it here, when the comparison is so simple. Compare to `isFunction`, which has a much more complicated definition.